packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeFormatService.ts

```diff
@@ -0,0 +1,566 @@
+        `Error in presentation definition while processing presentation exchange request: ${result[0].message} `
+      )
+    }
+  }
+
+  public async createPresentation(
+    agentContext: AgentContext,
+    options: FormatCreatePresentationOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofAttachmentFormat> {
+    if (!options.proofFormats.presentationExchange) {
+      throw Error('Presentation Exchange format missing while creating presentation in presentation exchange service.')
+    }
+
+    const requestPresentation = options.attachment.getDataAsJson<RequestPresentationExchangeOptions>()
+
+    const credential: IVerifiableCredential = options.proofFormats.presentationExchange.formats
+
+    const pex: PEXv1 = new PEXv1()
+
+    const subject: ICredentialSubject = credential.credentialSubject as ICredentialSubject
+
+    // Credential is allowed to be presented without a subject id. In that case we can't prove ownerhsip of credential
+    // And it is more like a bearer token.
+    // In the future we can first check the holder key and if it exists we can use that as the one that should authenticate
+    // https://www.w3.org/TR/vc-data-model/#example-a-credential-issued-to-a-holder-who-is-not-the-only-subject-of-the-credential-who-has-no-relationship-with-the-subject-of-the-credential-but-who-has-a-relationship-with-the-issuer
+    if (!subject?.id) {
+      throw new AriesFrameworkError(
+        'Credential subject missing from the selected credential for creating presentation.'
+      )
+    }
+
+    const didResolutionResult = await this.didResolver.resolve(agentContext, subject.id)
+
+    if (!didResolutionResult.didDocument) {
+      throw new AriesFrameworkError(`No did document found for did ${subject.id}`)
+    }
+
+    if (!didResolutionResult.didDocument?.authentication) {
+      throw new AriesFrameworkError(`No did authentication found for did ${subject.id} in did document`)
+    }
+
+    if (!didResolutionResult.didDocument?.verificationMethod) {
+      throw new AriesFrameworkError(`No did verification method found for did ${subject.id} in did document`)
+    }
+
+    const proofPurpose = IProofPurpose.authentication
+
+    // the signature suite to use for the presentation is dependant on the credentials we share.
+
+    // assertionMethod?: Array<string | VerificationMethod>
+
+    // 1. Get the key for this given proof purpose in this DID document
+    const keyId = didResolutionResult.didDocument[proofPurpose] as string[]
+
+    // get keys from the did document section containing the proof purpose
+
+    // 2. Map the Key Id to a key. Key contains publicKey and keyType attributes
+    const privateKey: Key = keyReferenceToKey(didResolutionResult.didDocument, keyId[0])
+
+    // 3. Use the retrieved key to determine proof type
+    const proofType = keyTypeToProofType(privateKey)
+
+    if (!proofType) {
+      throw new AriesFrameworkError(`Unsupported key type: ${privateKey.keyType}`)
+    }
```

Comment:
If you add the following method to the w3CCredentialService:

```
  public getProofTypeByVerificationMethodType(verificationMethodType: string): string {
    const suite = this.suiteRegistry.getByVerificationMethodType(verificationMethodType)

    if (!suite) {
      throw new AriesFrameworkError(`No suite found for verification method type ${verificationMethodType}}`)
    }

    return suite.proofType
  }
```

We can rewrite this to the following, which leverages more of the existing AFJ APIs:

```
    // 1. Get the verification method for this given proof purpose in this DID document
    let [verificationMethod] = didResolutionResult.didDocument.authentication
    if (typeof verificationMethod === 'string') {
      verificationMethod = didResolutionResult.didDocument.dereferenceKey(verificationMethod, ['authentication'])
    }

    const proofType = this.w3cCredentialService.getProofTypeByVerificationMethodType(verificationMethod.type)
```